packages/ddp-client/common/livedata_connection.js

```diff
@@ -550,6 +550,37 @@ export class Connection {
     }
     return this.apply(name, args, callback);
   }
+  /**
+   * @memberOf Meteor
+   * @importFromPackage meteor
+   * @alias Meteor.callAsync
+   * @summary Invokes an async method passing any number of arguments.
+   * @locus Anywhere
+   * @param {String} name Name of method to invoke
+   * @param {EJSONable} [arg1,arg2...] Optional method arguments
+   * @returns {Promise}
+   */
+  async callAsync(name /* .. [arguments] .. callback */) {
+    const args = slice.call(arguments, 1);
+    if (args.length && typeof args[args.length - 1] === 'function') {
+      throw new Error(
+        "Meteor.callAsync() does not accept a callback. You should 'await' the result, or use .then()."
+      );
+    }
+
+    return new Promise((resolve, reject) => {
+      DDP._CurrentMethodInvocation.set();
+      Meteor.bindEnvironment(() => {
+        this.applyAsync(name, args, (err, result) => {
+          if (err) {
+            reject(err);
+            return;
+          }
+          resolve(result);
+        });
+      })();
+    });
```

Comment:
This should be as easy as `return this.applyAsync(name, args);`. Is there any reason for not doing that?

Reply:
I added a comment in the code to explain this code. Here is the comment:

```
This is necessary because when you call a Promise.then, you're actually calling a bound function by Meteor.
     
This is done by this code https://github.com/meteor/meteor/blob/17673c66878d3f7b1d564a4215eb0633fa679017/npm-packages/meteor
     
So, without resting the context at this point, and then calling Meteor.bindEnvironment before calling
applyAsync, when you call a ".then()", like "Meteor.callAsync().then()", the global context (inside currentValues)
will be from the call of Meteor.callAsync(), and not the context after the promise is done.
     
This means that without this code if you call a stub inside the ".then()", this stub will act as a simulation
and won't reach the server.
```

TLDR: Without this, you can't use `Meteor.callAsync().then()`.